drivers/modem/gsm_ppp.c

```diff
@@ -944,6 +947,10 @@ static void gsm_configure(struct k_work *work)
 
 	LOG_DBG("Starting modem %p configuration", gsm);
 
+	if (gsm->modem_on_cb) {
+		gsm->modem_on_cb(NULL);
```

Comment:
Please store the user_data to gsm_modem struct and use that here instead of NULL.

Reply:
What about dev pointer, should I pass one of these 

<pre>
	const struct device *ppp_dev;
	const struct device *at_dev;
	const struct device *control_dev;
</pre> 
or null like here
<pre>gsm->modem_on_cb(NULL, gsm);</pre>

Reply:
Done.